src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -422,14 +476,214 @@ export class DeployCommand implements Command {
         );
         return parameterFilePath;
       }
+    } else if (result == this._none) {
+      return undefined;
+    } else {
+      this.outputChannelManager.appendToOutputChannel(
+        `Parameter file used in deployment -> ${result.data}`
+      );
+      return result.data;
+    }
+
+    return undefined;
+  }
+
+  private async handleMissingAndDefaultParams(
+    _context: IActionContext,
+    documentPath: string,
+    parameterFilePath: string | undefined,
+    template: string | undefined
+  ): Promise<[string, BicepUpdatedDeploymentParameter[]]> {
+    const bicepDeploymentParametersResponse: BicepDeploymentParametersResponse =
+      await this.client.sendRequest("workspace/executeCommand", {
+        command: "getDeploymentParameters",
+        arguments: [documentPath, parameterFilePath, template],
+      });
+
+    if (bicepDeploymentParametersResponse.errorMessage) {
+      throw new Error(bicepDeploymentParametersResponse.errorMessage);
+    }
+
+    const updatedDeploymentParameters: BicepUpdatedDeploymentParameter[] = [];
+
+    for (const deploymentParameter of bicepDeploymentParametersResponse.deploymentParameters) {
+      const paramName = deploymentParameter.name;
+      let paramValue = undefined;
+      if (deploymentParameter.isMissingParam) {
+        paramValue = await vscode.window.showInputBox({
+          title: `Parameter: ${paramName}`,
+          placeHolder: `Please enter value for parameter "${paramName}"`,
+        });
+      } else {
+        if (deploymentParameter.isExpression) {
+          paramValue = await this.selectValueForParameterOfTypeExpression(
+            _context,
+            paramName,
+            deploymentParameter.value
+          );
+        } else {
+          const options = {
+            title: `Parameter: ${paramName}`,
+            value: deploymentParameter.value,
+            placeHolder: `Please enter value for parameter "${paramName}"`,
+          };
+          paramValue = await vscode.window.showInputBox(options);
+        }
+      }
+
+      if (paramValue != undefined) {
```

Comment:
I'm surprised eslint isn't catching this.  You should always use !== and === instead of != and == in TypeScript.  It ensures no conversion takes place (similar to "is null" in c#).